lib/services/profile_S/profile_service.dart

```diff
@@ -57,5 +59,44 @@ class ProfileService {
       return null;
     }
   }
+
+  Future<String?> getProfileName(String userId) async {
+    FirebaseFirestore db = FirebaseFirestore.instance;
+
+    DocumentSnapshot profileSnapshot = await db.collection("profile_data").doc(userId).get(); //get the document
+    Map<String, dynamic> data = profileSnapshot.data() as Map<String, dynamic>; //map the overall data
+    Map<String, dynamic> userInfo = data['username'] as Map<String, dynamic>; //map the username data
+    // Return profile name if exists otherwise give an empty string
+    if (profileSnapshot.exists) {
+      return userInfo['profile_name'];
+    } else {
+      return '';
+    }
+  }
+
+  Future<void> editUsername(String username) async
+  {
+    try{
+      FirebaseFirestore db = FirebaseFirestore.instance;
+      final FirebaseAuth auth = FirebaseAuth.instance;
+      final currentUser = auth.currentUser;
+      if (currentUser != null) {
+        if (username.isNotEmpty) {
+          final data = { "username.profile_name" :username};
+          await db.collection("profile_data").doc(currentUser.uid).update(data);
+          await AuthService().getNextNumber();
+          int nextnumber = await AuthService().getNextNum();
+
+          if (nextnumber != 0) {
+            final number = { "username.unique_num" :nextnumber};
```

Comment:
Question: when a new profile is created, isn't there a unique number already assigned to the profile? If so, this would then assign a new unique number?

Reply:
So as far as I understand the unique number is not assigned when a profile is created or atleast not in all the cases. I could be wrong, but I didn't explicitly check for it. The good thing is that even though if there already was a unique number this would just replace that unique number, so I feel it wouldn't be too much of an issue, but can definitely be fixed or removed if needed.

Reply:
The default profile does assign a unique number to the profile. The only reason I'm a tiny bit concerned about this is, this will make our unique number go up a lot, and currently we have it to be 4 digits, so if it comes to 9999, we either have to reset it or just let it go up. I don't know which one we want to do

Reply:
I also definitely agree I thought about it too. Let me restructure the code a bit too hopefully prevent it from happening.